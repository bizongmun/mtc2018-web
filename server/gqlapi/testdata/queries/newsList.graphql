query {
  newsList {
    pageInfo {
      startCursor
      endCursor
      hasNextPage
      hasPreviousPage
    }
    edges {
      cursor
      node {
        ...NewsFragment
      }
    }
    nodes {
      ...NewsFragment
    }
  }
}

fragment NewsFragment on News {
  id
  date
  message
  messageJa
  link
}
