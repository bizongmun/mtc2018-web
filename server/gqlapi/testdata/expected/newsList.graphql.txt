--- newsList.graphql
query {
  newsList {
    pageInfo {
      startCursor
      endCursor
      hasNextPage
      hasPreviousPage
    }
    edges {
      cursor
      node {
        ...NewsFragment
      }
    }
    nodes {
      ...NewsFragment
    }
  }
}

fragment NewsFragment on News {
  id
  newsId
  date
  message
  messageJa
  link
}


--- Result
{"data":{"newsList":{"pageInfo":{"startCursor":null,"endCursor":"News:1","hasNextPage":false,"hasPreviousPage":false},"edges":[{"cursor":"News:2","node":{"id":"News:2","newsId":"2","date":"2018-09-28","message":"The list of exhibition booths is now up on our website!","messageJa":"展示ブースの内容を公開しました。","link":"https://mercari-tech-conf-2018.peatix.com"}},{"cursor":"News:1","node":{"id":"News:1","newsId":"1","date":"2018-09-04","message":"Tickets are on sale.","messageJa":"チケット販売をオープンしました","link":"https://mercari-tech-conf-2018.peatix.com"}}],"nodes":[{"id":"News:2","newsId":"2","date":"2018-09-28","message":"The list of exhibition booths is now up on our website!","messageJa":"展示ブースの内容を公開しました。","link":"https://mercari-tech-conf-2018.peatix.com"},{"id":"News:1","newsId":"1","date":"2018-09-04","message":"Tickets are on sale.","messageJa":"チケット販売をオープンしました","link":"https://mercari-tech-conf-2018.peatix.com"}]}}}

--- Result (formatted)
{
  "data": {
    "newsList": {
      "edges": [
        {
          "cursor": "News:2",
          "node": {
            "date": "2018-09-28",
            "id": "News:2",
            "link": "https://mercari-tech-conf-2018.peatix.com",
            "message": "The list of exhibition booths is now up on our website!",
            "messageJa": "展示ブースの内容を公開しました。",
            "newsId": "2"
          }
        },
        {
          "cursor": "News:1",
          "node": {
            "date": "2018-09-04",
            "id": "News:1",
            "link": "https://mercari-tech-conf-2018.peatix.com",
            "message": "Tickets are on sale.",
            "messageJa": "チケット販売をオープンしました",
            "newsId": "1"
          }
        }
      ],
      "nodes": [
        {
          "date": "2018-09-28",
          "id": "News:2",
          "link": "https://mercari-tech-conf-2018.peatix.com",
          "message": "The list of exhibition booths is now up on our website!",
          "messageJa": "展示ブースの内容を公開しました。",
          "newsId": "2"
        },
        {
          "date": "2018-09-04",
          "id": "News:1",
          "link": "https://mercari-tech-conf-2018.peatix.com",
          "message": "Tickets are on sale.",
          "messageJa": "チケット販売をオープンしました",
          "newsId": "1"
        }
      ],
      "pageInfo": {
        "endCursor": "News:1",
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": null
      }
    }
  }
}

